### DISCUSSION TITLE
Save row won't update relationship

### DISCUSSION THREAD

**Discussion by LucianHorrobin** [Budibase/budibase#14034] [Help] [answered]
This is my first post so please be kind.
I have two Budibase tables, Clients and Assets, that are in a many to many relationship and this works as expected in the Data view.
I am trying to build a client onboarding screen that links the client to the asset, this screen is Public as it will be embedded in my website and sent to prospective leads

![Onboarding screen](https://github.com/Budibase/budibase/assets/174016352/6b48d504-7ff4-4f95-9b72-11354d7d4968)

The screen is comprised of two form blocks and a submit button. The actions on the submit button are two save rows, one for each table.

![save row asset details](https://github.com/Budibase/budibase/assets/174016352/9cdf8d7b-24b0-4892-8f59-bcc861435b2a)

which updates the tables though dosen't link the entries Asset Owner/Client Assets column in the table

![Clients table](https://github.com/Budibase/budibase/assets/174016352/cd433541-1c96-44fc-a6e5-dadc3585d1f0)
![Assets table](https://github.com/Budibase/budibase/assets/174016352/750a51e3-3770-4d6d-9f17-6e525dc01330)

I've tried every combintaion of actions, id's, save row's and repeaters i can find other help for, though just can't seem to get it to work.

Any help on this would be muchly appreciated.

**Comment by mikesealey** [comment, marked as answer]
Hi @LucianHorrobin 

This sounds like a great use of Budibase! 

Looking at your component tree, it seems like you're using a Form component, rather than a Form Block component. The difference being the Form Block Component is pretty rigid, straight out of the box it will have a lot of things configured for you, whereas the Form Component lets you build a form out however you want, but there's a bit more work that needs to go into it. This is actually good news though, it gives us more scope to do what we want.

In your Submit button, add two Save Row actions - the first should save your Client information to the client table
(Mine looks like this)
![image](https://github.com/Budibase/budibase/assets/39269380/ff525e9f-71e6-4145-875d-0d648df4c769)

Your second Save Row action should save your asset, with a little nuance though
![image](https://github.com/Budibase/budibase/assets/39269380/28047161-86ab-4cb9-96f3-98ac085a5993)

In the Asset_Owner column (the relationship column) we use the `_id` of the Client row we saved in the first Save Row action.
`return $("Action 1.Saved row._id")` 

Lastly, there's some cleanup to do - two Clear Form actions tidy up our fields, and you might want to go into one or both of the Save Row actions to disable the default notification to prevent the `(2)` from appearing in the success message.
![image](https://github.com/Budibase/budibase/assets/39269380/847fbe52-f38b-410b-8070-72bf62dd9393)

In my example, I've also added two Table Components side-by-side underneath the form, and I can see the relationship is immediately populated when saving the rows.

![image](https://github.com/Budibase/budibase/assets/39269380/0627e500-ac1f-4bc6-b35a-3f74800c856d)

I've also attached an app-export of my work so you can import it and have a closer look, if you feel that would be beneficial. If you're unsure, you can learn how to import an app[here.](https://docs.budibase.com/docs/export-and-import-apps)
[14034-export-1719498114322.tar.gz](https://github.com/user-attachments/files/16015540/14034-export-1719498114322.tar.gz)

You might also find answers and inspiration from other members of the Budibase community in our [Discord Server](https://discord.gg/ZepTmGbtfF)

I hope that helps solve your puzzle. Let us know if you need any clarifications!

**Comment by LucianHorrobin** [comment]
OMG the ._id on the action, I looked back at all the doccumentation and there it was!
Mike, thank you so much.
I ended up using a standard binding, which worked a treat.
